modules/signals/src/signal-store-models.ts

```diff
@@ -15,12 +15,11 @@ export type SignalStoreSlices<State> = IsKnownRecord<
     }
   : {};
 
-export type SignalStore<FeatureResult extends SignalStoreFeatureResult> =
+export type SignalStoreProps<FeatureResult extends SignalStoreFeatureResult> =
   Prettify<
     SignalStoreSlices<FeatureResult['state']> &
       FeatureResult['signals'] &
-      FeatureResult['methods'] &
-      SignalStateMeta<Prettify<FeatureResult['state']>>
+      FeatureResult['methods']
```

Comment:
A workaround for this TS bug is to always put a unique symbol inside of the named type.

**Before:**

The `STATE_SIGNAL` was appended to the dynamic `SignalStore` type by prettifying the intersection.

```ts
const CounterStore = signalStore(withState({ count: 0 }));
// type: Type<{ count: Signal<number>; [STATE_SIGNAL]: WritableSignal<{ count: number }> }>
```

**After:**

The `STATE_SIGNAL` is always exported as a part of the `StateSignal` type.

```ts
const CounterStore = signalStore(withState({ count: 0 }));
// type: Type<{ count: Signal<number>; } & StateSignal<{ count: number }>>
```

Reply:
Makes sense. 

`StateSignal` has to be added to the public APIs `index.ts` as well, right?

Reply:
It's not needed. We can keep it private.